open-sphere-plugins/arcgis/src/main/java/io/opensphere/arcgis2/envoy/ArcGISLayerListEnvoy.java

```diff
@@ -250,6 +360,18 @@ else if ((serverIndex = split.indexOf("FeatureServer")) >= 0)
             {
                 builder.setPath(split.subList(servicesIndex + 1, serverIndex));
             }
+            else if ((serverIndex = split.indexOf("GPServer")) >= 0)
```

Comment:
I think this if/else block is getting long enough it's probably time to refactor it to a for loop that iterates over an array or something.

Reply:
Yea, probably.